cluster-operator/src/test/java/io/strimzi/operator/cluster/FeatureGatesTest.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster;
+
+import io.strimzi.operator.common.InvalidConfigurationException;
+import io.strimzi.test.annotations.ParallelSuite;
+import io.strimzi.test.annotations.ParallelTest;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@ParallelSuite
+public class FeatureGatesTest {
+    @ParallelTest
+    public void testFeatureGates()  {
+        assertThat(new FeatureGates("+ControlPlaneListener").controlPlaneListenerEnabled(), is(true));
+        assertThat(new FeatureGates("  +ControlPlaneListener    ").controlPlaneListenerEnabled(), is(true));
+        assertThat(new FeatureGates("-ControlPlaneListener").controlPlaneListenerEnabled(), is(false));
+    }
+
+    @ParallelTest
+    public void testEmptyFeatureGates()  {
+        assertThat(new FeatureGates(null).controlPlaneListenerEnabled(), is(false));
+        assertThat(new FeatureGates("").controlPlaneListenerEnabled(), is(false));
+        assertThat(new FeatureGates(" ").controlPlaneListenerEnabled(), is(false));
+        assertThat(new FeatureGates("    ").controlPlaneListenerEnabled(), is(false));
+    }
+
+    @ParallelTest
+    public void testDuplicateFeatureGate() {
+        InvalidConfigurationException e = assertThrows(InvalidConfigurationException.class, () -> new FeatureGates("+ControlPlaneListener,+ControlPlaneListener"));
```

Comment:
Could you add the test setting `+Gate` and `-Gate`?

Reply:
Added.